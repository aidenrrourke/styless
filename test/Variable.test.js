import React from 'react';
import renderer from 'react-test-renderer';
import styled from 'styled-components';
import 'jest-styled-components';

test( 'Takes the color from props', () => {
	const Div = styled.div`
		background-color: chocolate;
		color: @primary;
	`;
	const tree = renderer.create( <Div primary="red"/> ).toJSON()
	expect( tree ).toMatchSnapshot();
} );

test( 'Replaces single variable in linear-gradient', () => {
	const Div = styled.div`
		background: linear-gradient(@primary 0%, darken(@primary, 5%) 100%);
	`;
	const tree = renderer.create( <Div primary="red"/> ).toJSON()
	expect( tree ).toMatchSnapshot();
} );