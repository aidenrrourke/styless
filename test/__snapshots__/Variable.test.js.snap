// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Replaces single variable in linear-gradient 1`] = `
.c0 {
  background: linear-gradient(red 0%,darken(red,5%) 100%);
}

<div
  className="c0"
/>
`;

exports[`Takes the color from props 1`] = `
.c0 {
  background-color: chocolate;
  color: red;
}

<div
  className="c0"
/>
`;
